samtranslator/model/api/api_generator.py

```diff
@@ -218,23 +219,26 @@ def _construct_api_domain(self, rest_api):
         Constructs and returns the ApiGateway Domain and BasepathMapping
         """
         if self.domain is None:
-            return None, None
+            return None, None, None
 
         if self.domain.get('DomainName') is None or \
            self.domain.get('CertificateArn') is None:
             raise InvalidResourceException(self.logical_id,
                                            "Custom Domains only works if both DomainName and CertificateArn"
                                            " are provided")
 
-        logical_id = logical_id_generator.LogicalIdGenerator("", self.domain).gen()
+        self.domain['ApiDomainName'] = "{}{}".format('ApiGatewayDomainName',
```

Comment:
I don't remember from the last PR- is it possible to reference the domain resource by using <ApiName>.Domain? If not, we might want to add support for that in a future PR for this feature.

Also, I'm still not sure if I like this name but can't think of a better one so it will probably stick 👍

Reply:
Yes, that change is in this PR.

Yeah, naming is a hard one. Let me know if you come up with something better.